Makefile

```
@@ -154,6 +154,26 @@ test-conformance: $(BIN)/conformance_test_runner $(BUILD)/protobuf-conformance
 		--text_format_failure_list packages/protobuf-conformance/conformance_failing_tests_text_format.txt \
 		packages/protobuf-conformance/bin/conformance_esm.js
 
+TS_VERSIONS = 4.1.2 \
+			  4.2.4 \
+			  4.3.5 \
+			  4.4.4 \
+			  4.5.2 \
+			  4.6.4 \
+			  4.8.1-rc
+
+.PHONY: test-ts-compat
+test-ts-compat: $(GEN)/protobuf-test node_modules 
+	@for number in $(TS_VERSIONS) ; do \
+		formatted=$$(echo "$${number}" | sed -r 's/[\.]/_/g'); \
+		dirname=packages/protobuf-test/typescript ; \
+		echo "Testing TypeScript `node_modules/ts$$formatted/bin/tsc --version`" ; \
+		node_modules/ts$$formatted/bin/tsc -p $$dirname/tsconfig.$${formatted}.json --noEmit ; \
+		if [ "$$?" -ne 0 ]; then \
+			exit 1 ; \
+	  	fi ; \
```

I would have expected a non-zero exit code from `tsc` to exit the loop. Turns out `-e` has some [surprising behavior](http://mywiki.wooledge.org/BashFAQ/105).

What do you think about simply using `||`?

```suggestion
		node_modules/ts$$formatted/bin/tsc -p $$dirname/tsconfig.$${formatted}.json --noEmit || exit; \
```